ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Jenkins_User/tasks/macos_autologon.yml

```diff
@@ -0,0 +1,67 @@
+# To run UI tests on macOS, macOS' UI server needs to be running. The UI
+# server is only active after a user has logged into their account and cannot
+# be started on demand like Xvfb. The only option is to enable auto logon for
+# the user running the tests which is "jenkins" in our case. Xvfb cannot be
+# used because macOS uses Quartz to render its UI, not X.
+#
+# The security implications of enabling auto logon are as follows:
+#
+# * Anybody with physical access does not have to authenticate to perform actions
+#   as "jenkins" (only if macOS is running on a physical machine).
+# * Any kind of remote login (screen sharing, SSH) needs the same authentication
+#   as usual (except it was disabled separately, of course).
+#
+# If you want to minimize the security risks associated with auto logon,
+# run the unlocked macOS in a Parallels VM (other types of VM might support
+# this, too) and configure Parallels to automatically open a window when the
+# VM starts. Then, the host machine can remain locked all the time an protect
+# the VMs from direct physical access.
+#
+# Apart from running the playbook, some manual configuration is needed: In
+# "System Preferences", go to "Security" > "Privacy" and click on
+# "Accessibility". Add whatever program runs the UI test (at AdoptOpenJDK, it's
+# "sshd-keygen-wrapper") and allow it to control the computer.
+---
+- name: Disable Screen Saver
+  osx_defaults:
+    domain: com.apple.screensaver
+    key: idleTime
+    type: int
+    value: 0
+    host: "currentHost"
+    state: present
+
+- name: Disable Screen Lock
+  shell: sysadminctl -screenLock off -password {{ ansible_become_pass }}
```

Comment:
Is there a variable for it, like `{{ Administrator password}}`? Using sudo does not help.

Reply:
No - the ansible playbooks do not currently have a way to access machine-specific information like that, although it could be set as a variable. Perhaps we could only run that role if the variable is set, and if not put out a warning message saying it should be done manually?

Reply:
I'd normally add those to the inventory and encrypt the values with Ansible Vault. We have the Vendor_Files concept: We might be able to add the variables there. Ultimately, we have to explain to third party users how do replicate those, anyway. And we'd have an integrated solution that would work with Ansible Tower.

Reply:
We're not currently using ansible vault, so for now I think making it optional, and allowing the user to pass it in if they are willing to give it to ansible, or do that part of the setup manually if required seems to be the best approach for now to get something that works.

Reply:
I wonder if we could use the Zeus user for this? We could assign a password the machine and run the task that way